graph/src/components/store/entity_cache.rs

```diff
@@ -209,11 +209,59 @@ impl EntityCache {
             id_is_bytes: id_is_bytes,
         };
 
+        // Get the entities from the store and insert it into the cache if it's not already there
         let entities = self.store.get_derived(&query)?;
-        entities.iter().for_each(|(key, e)| {
-            self.current.insert(key.clone(), Some(e.clone()));
+
+        entities.iter().for_each(|(k, v)| {
+            // Only insert to the cache if it's not already there
+            if !self.current.contains_key(k) {
+                self.current.insert(k.clone(), Some(v.clone()));
+            }
+        });
+
+        // Insert `None` for entities that are not in the store and not in the current cache
+        // but are present in updates or handler_updates
+        // The updates will be applied before returning the entities
+        self.updates.iter().for_each(|(k, _)| {
+            if !self.current.contains_key(k) {
+                self.current.insert(k.clone(), None);
+            }
         });
-        let entities: Vec<Entity> = entities.values().cloned().collect();
+
+        self.handler_updates.iter().for_each(|(k, _)| {
+            if !self.current.contains_key(k) {
+                self.current.insert(k.clone(), None);
+            }
+        });
+
+        // Retrieve the derived entities from the cache by filtering the cached entities.
+        // Include entities if either a matching value is found in the specified entity field
+        // as indicated by `query`, or if the entity is None.
+        let entities = self.current.iter().filter(|(key, entity)| {
+            key.entity_type == query.entity_type
+                && entity.as_ref().map_or(true, |e| {
+                    e.get(&query.entity_field.to_camel_case())
```

Comment:
Why is `to_camel_case` needed here?

Reply:
query.entity_field is in snake case and the same field is in snake case in the type `Entity`

Reply:
Ugh .. that's pretty wild. Definitely an oversight when we first merged this. Stuff called `_field` should use GraphQL names.

Reply:
Ah makes sense, there was a bug that i fixed for the derived loaders and i remember the fix was using to_snake_case() somewhere in the code.

Reply:
https://github.com/graphprotocol/graph-node/issues/4682

That happened because the store expected snake_case and needed to convert it, I've changed this back and only call `to_snake_case` at the point of constructing the sql query